### DISCUSSION TITLE
Catch a mistake, win a prize!

### DISCUSSION THREAD

**Discussion by koushikkothagal** [koushikkothagal/java-backend-bootcamp#11] [Announcements]
Announcing the "bug bounty" programs on the study material.

Catch any factual, conceptual, theoretical or code errors in the study material and you will get rewarded for it!

- Catch 3 issues: You get a free 1:1 consultation call with me
- Catch 5 issues: You get a free mock interview session with me with feedback
- Catch 7 issues: Free 6 months of All Access Pass
- Catch 10 issues: Free 2 years of the All Access Pass

**Comment by Bangosushi** [comment]
Not sure if it's a mistake but I would like to question the ["Generalization with Interfaces"](https://koushikkothagal.github.io/java-backend-bootcamp/docs/java-backend-bootcamp/java-collections/generalization-with-interfaces) page being in the ["Java Collections"](https://koushikkothagal.github.io/java-backend-bootcamp/docs/java-collections) topic. 

I'd argue it should probably be more suited with the other pages describing inheritance in Java in the ["Java Basics"](https://koushikkothagal.github.io/java-backend-bootcamp/docs/java-basics) topic.

**Reply by koushikkothagal** [reply]
Not a mistake, but a fair question @Bangosushi . The reason I put it there was to illustrate the design of the Collection APIs and how it affects our usage (Example: use `List` variables instead of `ArrayList`). I think it's useful to have as a refresher when someone new to Java reads the section on Collections.

**Comment by iamguptavishal** [comment]
Blog page - https://koushikkothagal.github.io/java-backend-bootcamp/docs/java-backend-bootcamp/java-collections/introduction

Says Collection(I) is a root interface for List(I), Set(I) and Map(I).
I have already raised pull request, as Collection(I) is not a root for Map(I)
